panels/sharing/cc-sharing-panel.c

```diff
@@ -660,6 +660,31 @@ cc_sharing_panel_setup_media_sharing_dialog (CcSharingPanel *self)
                                            WID ("media-sharing-status-label"));
 }
 
+#define DMI_PRODUCT_NAME "/sys/class/dmi/id/product_name"
+
+/* We don't use the "Computer Name" here since that is really the hostname
+ * which is more of an internal detail. */
+static gchar *
+get_display_computer_name (void)
```

Comment:
Instead of the product name, which is typically garbage, we should use the hostname, since it's also editable from the Details panel.
There's a nice dbus API for that; I think it should be possible to use it directly from the python script that writes the service file, so that we don't need to pass the name parameter over through the API.

At that point, the `Discoverable` property could be made read-write, since it would not need the name anymore, whose state is held separately by `hostnamectl`.

Reply:
I suppose at the time we enable the companion app services we would want to read the hostname. Is that right?

Reply:
Yeah, that sounds right; I was thinking you could read it directly from the python script that writes the service file